### PR TITLE
TagBox - improve items selection if last DataSource request is in progress (T978877)

### PR DESCRIPTION
**Pull request by alexander-kotov-dx**

### CODE REVIEW

File: js/ui/selection/selection.strategy.standard.js [hunk 1]
```diff
@@ -369,16 +397,25 @@ export default SelectionStrategy.inherit({
...
+    isItemDataSelected: function(itemData, checkPending) {
         const key = this.options.keyOf(itemData);
-        return this.isItemKeySelected(key);
+        return this.isItemKeySelected(key, checkPending);
     },
 
-    isItemKeySelected: function(key) {
-        const keyHash = this._getKeyHash(key);
-        const index = this._indexOfSelectedItemKey(keyHash);
+    isItemKeySelected: function(key, options) {
```
Comment (DokaRus) [inline js/ui/selection/selection.strategy.standard.js]: Why are the arguments inconsistent? In the `isItemDataSelected` method second arg is a Boolean, in the `isItemKeySelected` method it is an object

File: js/ui/selection/selection.strategy.standard.js [hunk 2]
```diff
@@ -369,16 +397,25 @@ export default SelectionStrategy.inherit({
...
+        let result;
+        const checkPending = options?.checkPending;
+        if(checkPending && this._lastRequestData && this._requestInProgress()) {
+            result = this._lastRequestData.addedItems && this._lastRequestData.addedItems.indexOf(key) !== -1;
+        }
```
Comment (DokaRus) [inline js/ui/selection/selection.strategy.standard.js]: Let's extract "data mining" :)
```suggestion
        let result = this._getRequestResult(checkPending);
        
        if(!result) {
        ...
        const checkPending = options?.checkPending;
        if(checkPending && this._lastRequestData && this._requestInProgress()) {
            result = this._lastRequestData.addedItems && this._lastRequestData.addedItems.indexOf(key) !== -1;
        }
```

**Review by ksercs — DISMISSED** [review]

**Review by Zedwag — DISMISSED** [review]

**Review by DokaRus — DISMISSED** [review]

File: js/ui/selection/selection.strategy.standard.js [hunk 3]
```diff
@@ -369,16 +406,21 @@ export default SelectionStrategy.inherit({
...
         const key = this.options.keyOf(itemData);
-        return this.isItemKeySelected(key);
+        return this.isItemKeySelected(key, options);
     },
 
-    isItemKeySelected: function(key) {
-        const keyHash = this._getKeyHash(key);
-        const index = this._indexOfSelectedItemKey(keyHash);
+    isItemKeySelected: function(key, options) {
+        let result = this._isItemSelectionInProgress(key, options?.checkPending);
```
Comment (DokaRus) [inline js/ui/selection/selection.strategy.standard.js]:
```suggestion
    isItemKeySelected: function(key, options = {}) {
        let result = this._isItemSelectionInProgress(key, options.checkPending);
```

File: js/ui/selection/selection.strategy.standard.js [hunk 4]
```diff
@@ -369,16 +406,21 @@ export default SelectionStrategy.inherit({
         this._updateRemovedItemKeys(keys, oldSelectedKeys, oldSelectedItems);
     },
 
-    isItemDataSelected: function(itemData) {
+    isItemDataSelected: function(itemData, options) {
```
Comment (DokaRus) [inline js/ui/selection/selection.strategy.standard.js]:
```suggestion
    isItemDataSelected: function(itemData, options = {}) {
```

**Review by Zedwag — APPROVED** [review]

**Pull request merged**